plugin/documents.py

```diff
@@ -680,38 +672,35 @@ def _on_signature_help(
         point: int,
         language_map: MarkdownLangMap | None
     ) -> None:
-        self._sighelp = SigHelp.from_lsp(response, language_map)
-        if self._sighelp:
-            content = self._sighelp.render(self.view)
-
-            def render_sighelp_on_main_thread() -> None:
-                if self.view.is_popup_visible():
-                    self._update_sighelp_popup(content)
-                else:
-                    self._show_sighelp_popup(content, point)
-
-            sublime.set_timeout(render_sighelp_on_main_thread)
+        sighelp = SigHelp.from_lsp(response, language_map)
+        if sighelp:
+            # Render on main thread.
+            sublime.set_timeout(lambda: self._show_sighelp_popup(sighelp, point))
+        elif self._sighelp:
```

Comment:
These if elif conditions are confusing me.
if sighelp, then show popup
elif self._sighelp, then hide popup 

on first glance, 
sighelp and self._sighelp look like they store the same data, 
but cause opposite actions